### DISCUSSION TITLE
WhatsApp | System Design

### DISCUSSION THREAD

**Discussion by giscus** [karanpratapsingh/portfolio#34] [Comments]
# WhatsApp | System Design

In this tutorial, we will learn how to design WhatsApp

https://www.karanpratapsingh.com/courses/system-design/whatsapp

**Comment by katej03** [comment]
Amazing tutorial, **thanks** so much

**Reply by karanpratapsingh** [reply]
Thank you @katej03!

**Comment by JaydeepMotisariya** [comment]
Love it. Very detailed documentation. Thank you so much

**Reply by karanpratapsingh** [reply]
Thank you @JaydeepMotisariya!

**Comment by williamjaxon32** [comment]
WhatsApp plus uses an Ejabberd server (more on that later) to communicate with its clients using a highly modified version of XMPP. An SSL socket is opened between the client and[ WhatsApp plus](https://gbfunda.com/whatsapp-plus/) servers via the XMPP protocol. To retrieve the messages, the client must open or reconnect to this socket until the server queues the messages.

**Comment by michaelmcguirk** [comment]
Thanks for the write-up. You have a small error in the traffic calculation: "50 million×20 messages=2 billion/day" should be 1billion/day?

**Reply by karanpratapsingh** [reply]
Nice catch! Thanks

**Comment by bagherani** [comment]
It's worth noting that they keep the messages for 30 days in their database for facilitating message delivery and of course, save costs.
So maybe instead of 10 years, we can assume 30 days, would be 306TB storage needed.
